### ISSUE TITLE
Illegal Access - Default Deny Field Access Request

### ISSUE THREAD

**Issue by GedMarc** [FasterXML/jackson-databind#3276]
All Reflective Access into JDK Internal's modules have been disabled by default in JDK 16 and above,
The ability to override this in JDK 17 has been removed

**Describe the solution you'd like**
Deny all Field access to any classes located in the jdk.internal module library (java.base/java.util/java.logging etc, making sure the implementation is safe for JDK 8 as well), 

Use properties (get/set/is) *only* for all and any classes located in these closed packages 
* Date
* LocalDateTime and the entire suite
* java.sql.xxx
* java.util.concurrent.atomic.xxxxx

Initial thoughts are :  
* Implement the auto detect function to use None for fields by default for all classes located under the java. namespace, and javax namepsace
* perhaps a package-info mixin, not sure?

**Usage example**
```
class xMe
{
private LocalDate failOnIllegalAccessFieldAccess;
}
```

**Additional context**
The change shouldn't impact anything backwards compatible, as the same fields used for serialization and deserialization do exists as properties in all the affected classes

**Comment by cowtowncoder** [comment]
Yes, this has been a problem looming in the distance for a while.

Skipping auto-detection of `Field`s within `java.` and `javax.` might make sense, and add minor performance optimization too. Some questions include:

1. Is anyone making use of mix-in annotations to force use of some Fields within classes here?
2. I think that the actual issue is not auto-detection but forced access (trying to make `Field`s settable/gettable via Reflection) -- perhaps it would make sense to start preventing access overrides, or, maybe, filtering these out during processing. But then again, if they are useless, why even carry them around I guess.

Also, I think one good way to make some progress would be to have some (failing) tests for JDK 17, JDK type(s) that trigger this problem.

**Comment by cowtowncoder** [comment]
Realized there is #3117 which seems to cover some of this, to be included in 2.13.0.

**Comment by cowtowncoder** [comment]
Will consider solved by #3117.

May re-file with further improvement ideas, suggestions, for 3.x versions.
